usecases/scenario_testrun_usecase.go

```diff
@@ -0,0 +1,69 @@
+package usecases
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/pure_utils"
+	"github.com/checkmarble/marble-backend/repositories"
+	"github.com/checkmarble/marble-backend/usecases/executor_factory"
+	"github.com/checkmarble/marble-backend/usecases/security"
+	"github.com/pkg/errors"
+)
+
+type ScenarioTestRunUsecase struct {
+	transactionFactory             executor_factory.TransactionFactory
+	executorFactory                executor_factory.ExecutorFactory
+	scenarioPublicationsRepository repositories.ScenarioPublicationRepository
+	enforceSecurity                security.EnforceSecurityScenario
+	clientDbIndexEditor            clientDbIndexEditor
+	repository                     repositories.ScenarioTestRunRepository
+}
+
+func (usecase *ScenarioTestRunUsecase) ActivateScenarioTestRun(ctx context.Context,
+	organizationId string,
+	input models.ScenarioTestRunInput,
+) (models.ScenarioTestRun, error) {
+	exec := usecase.executorFactory.NewExecutor()
+	// we should not have any existing testrun for this scenario
+	existingTestrun, err := usecase.repository.GetByScenarioIterationID(ctx, exec, input.ScenarioIterationId)
+	if err != nil {
+		return models.ScenarioTestRun{}, errors.Wrap(err,
+			"error while fecthing entries to find an existing testrun")
+	}
+	if existingTestrun.ScenarioIterationId != "" && existingTestrun.Status == models.Up {
```

Comment:
ok for the idea, but see comment in the repo for the actual implem of GetByScenarioIterationID